bottlerocket/agents/src/bin/ec2-resource-agent/ec2_provider.rs

```diff
@@ -340,35 +344,52 @@ fn userdata(
     endpoint: &Option<String>,
     certificate: &Option<String>,
     cluster_dns_ip: &Option<String>,
+    custom_user_data: &Option<CustomUserData>,
     memo: &ProductionMemo,
 ) -> ProviderResult<String> {
-    Ok(match cluster_type {
-        ClusterType::Eks => base64::encode(format!(
-            r#"[settings.updates]
-ignore-waves = true
-    
-[settings.kubernetes]
-api-server = "{}"
-cluster-name = "{}"
-cluster-certificate = "{}"
-cluster-dns-ip = "{}""#,
-            endpoint
-                .as_ref()
-                .context(memo, "Server endpoint is required for eks clusters.")?,
-            cluster_name,
-            certificate
-                .as_ref()
-                .context(memo, "Cluster certificate is required for eks clusters.")?,
-            cluster_dns_ip
-                .as_ref()
-                .context(memo, "Cluster DNS IP is required for eks clusters.")?,
-        )),
-        ClusterType::Ecs => base64::encode(format!(
-            r#"[settings.ecs]
-cluster = "{}""#,
-            cluster_name,
-        )),
-    })
+    let userdata = match custom_user_data {
+        Some(CustomUserData::Replace { encoded_userdata }) => encoded_userdata.to_string(),
+        _ => {
+            let temp_userdata = match cluster_type {
+                ClusterType::Eks => base64::encode(format!(
```

Comment:
Can you move these string formats out into two functions so that the branching logic here is easier to read?

```rust
fn default_eks_userdata(x, y, z) -> String { todo!() }
fn default_ecs_userdata(x) -> String { todo!() }
```